pennylane_qiskit/devices.py

```diff
@@ -171,20 +191,25 @@ def apply(self, operation, wires, par):
         else:
             raise ValueError("The operation is not of an expected type. This is a software bug!")
 
-    def pre_expval(self):
-        compile_backend = self._provider.get_backend(self.compile_backend)  # type: BaseBackend
-
-        for qr, cr in zip(self._reg, self._creg):
-            measure(self._circuit, qr, cr)
-
+    def _compile_and_execute(self):
+        compile_backend = self.compile_backend if self.compile_backend is not None else self.backend  # type: BaseBackend
         compiled_circuits = qiskit.compiler.transpile(self._circuit, backend=compile_backend)
         qobj = qiskit.compiler.assemble(experiments=compiled_circuits, backend=compile_backend, shots=self.shots)
-        backend = self._provider.get_backend(self.backend)  # type: BaseBackend
+        backend = self.backend  # type: BaseBackend
 
         try:
-            if isinstance(backend, AerBackend) and (isinstance(self, BasicAerQiskitDevice) or
-                                                    isinstance(self, AerQiskitDevice)):
-                self._current_job = backend.run(qobj, noise_model=self._noise_model)
+            if isinstance(backend, AerBackend) and isinstance(self, AerQiskitDevice):
+                self._current_job = backend.run(qobj, noise_model=self._noise_model, backend_options=self.kwargs)
+
+            elif isinstance(backend, QasmSimulatorPy) and isinstance(self, BasicAerQiskitDevice):
+                self._current_job = backend.run(qobj, backend_options=self.kwargs)
+
+            elif isinstance(backend, StatevectorSimulatorPy) and isinstance(self, BasicAerQiskitDevice):
+                self._current_job = backend.run(qobj, backend_options=self.kwargs)
+
+            elif isinstance(backend, UnitarySimulatorPy) and isinstance(self, BasicAerQiskitDevice):
+                self._current_job = backend.run(qobj, backend_options=self.kwargs)
```

Comment:
Apart from `AerBackend`, where the noise model option is available, is it possible to merge all of these into one else statement?
```python
else:
    self._current_job = backend.run(qobj, backend_options=self.kwargs)
```

Reply:
You could, I don't want to though. I expect changes and wanted to 'document' that we have the ability to go into detail with any Device. If you insist we can change this of course!